src/Agent.Worker/Release/ReleaseFileSystemManager.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    [ServiceLocator(Default = typeof(ReleaseFileSystemManager))]
+    public interface IReleaseFileSystemManager : IAgentService
```

Comment:
Consider:
I'm guessing this is here for testability. I've shyed away from implementing such a class so far. It's often more trouble than it's worth and easier to write tests that actually hit disk (which is OK in this repo).

I briefly scanned where this is called from. It looks like you have control over the directories so if you add unit tests that hit disk, you could clean it up at the end of your test.